docs/DEBUGGING.md

```diff
@@ -0,0 +1,56 @@
+# Tips for debugging in `jq`
+
+`jq` comes with a handy [`debug`][debug] filter.
+Use it while you are developing your exercise solutions to inspect the data that is currently in the jq pipline.
+
+`debug` print a "debug array" to **stderr**.
+The first element of the array is the string "DEBUG:".
+The second element depends on how you invoke `debug`:
+
+1. the zero-arity debug function puts a compact representation of the input into the array:
+
+    ```sh
+    jq -n '[11, 22, 33] | debug | map(. * 2)'
+    ```
+
+    outputs
+
+    ```none
+     ["DEBUG:",[11,22,33]]
+     [
+       22,
+       44,
+       66
+     ]
+    ```
+
+2. the one-arity version pipes the input through the given filter, prints the debug message, and outputs the input unchanged:
```

Comment:
You might want to mention that both versions outputs the input unchanged. Possibly separate from discussing the differences between the versions.